subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -137,8 +139,22 @@ static struct nrf_cloud_pgps_prediction *get_cached_prediction(off_t off)
 	if (prediction_buf_flash_offset != off) {
 		int err;
 
+		LOG_DBG("Checking if flash_accessible...");
+		err = k_sem_take(&flash_accessible, K_NO_WAIT);
```

Comment:
would an atomic variable be sufficient for this use case?

Reply:
No, because the download codepath which leads to writing to flash can be called from the download_client in the background, while various functions which might want to retrieve a prediction from flash can be called by the application in a thread or worker function.

Reply:
i meant specifically an `atomic_t` type.
https://docs.zephyrproject.org/3.0.0/reference/kernel/other/atomic.html

seems like it would be sufficient, you aren't using any of the wait features of the semaphore.

Reply:
I suppose if I use the `atomic_test_and_set_bit()` or `atomic_cas()` functions.

Reply:
But, the pre-existing pgps_active semaphore works the same way as this new one, so I think I'd prefer to leave both as semaphores and then look at replacing them in a future PR.

Reply:
i think `atomic_read` to get the value, then `atomic_set` to set would be more straightforward.

Reply:
oh, `atomic_cas` would be useful.

Reply:
The need for either of the functions I mentioned is to make sure that, between an `atomic_read` and an `atomic_set`, another thread doesn't come in and do the same, and steal access from the first thread.  Both would think it was OK to access the flash.

Reply:
Ah, I didn't realize `atomic_set` returns the previous value, so you could probably use that instead of `atomic_test_and_set_bit` or `atomic_cas`.

Reply:
yes.  `atomic_cas` replaces the take, and `atomic_set` replaces the give.

Reply:
If you really want me to do this, I'm not comfortable trying to get this fix in for NCS 2.2.0 (assuming tomorrow is the deadline).

Reply:
it doesn't need to be done... the semaphore just seemed overkill considering how it was used.
i would be more concerned about ensuring the semaphore is always given after a take.

Reply:
I have covered that case in my latest commit for http(s) downloads.  But there is another transport option used by LWM2M which I do not cover completely.  I'm trying to find a solution for that.